

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" />
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  
  <title>All configurable options &mdash; seq2science  documentation</title>
  

  
  <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />

  
  

  
  

  

  
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
        <script src="../_static/language_data.js"></script>
    
    <script type="text/javascript" src="../_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Changelog" href="CHANGELOG.html" />
    <link rel="prev" title="Per rule explanation" href="all_rules.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../index.html" class="icon icon-home"> seq2science
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="gettingstarted.html">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="workflows.html">Workflows</a></li>
<li class="toctree-l1"><a class="reference internal" href="results.html">Using the results</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="extensive_docs.html">Extensive docs</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="cli.html">Command line interface</a></li>
<li class="toctree-l2"><a class="reference internal" href="all_rules.html">Per rule explanation</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="#">All configurable options</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#general">general</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#samples">samples</a></li>
<li class="toctree-l4"><a class="reference internal" href="#rule-dir">rule_dir</a></li>
<li class="toctree-l4"><a class="reference internal" href="#result-dir">result_dir</a></li>
<li class="toctree-l4"><a class="reference internal" href="#genome-dir">genome_dir</a></li>
<li class="toctree-l4"><a class="reference internal" href="#log-dir">log_dir</a></li>
<li class="toctree-l4"><a class="reference internal" href="#benchmark-dir">benchmark_dir</a></li>
<li class="toctree-l4"><a class="reference internal" href="#fqext1">fqext1</a></li>
<li class="toctree-l4"><a class="reference internal" href="#fqext2">fqext2</a></li>
<li class="toctree-l4"><a class="reference internal" href="#fqsuffix">fqsuffix</a></li>
<li class="toctree-l4"><a class="reference internal" href="#cpulimit">cpulimit</a></li>
<li class="toctree-l4"><a class="reference internal" href="#email">email</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#alignment-general">alignment general</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#trimmed-dir">trimmed_dir</a></li>
<li class="toctree-l4"><a class="reference internal" href="#qc-dir">qc_dir</a></li>
<li class="toctree-l4"><a class="reference internal" href="#final-bam-dir">final_bam_dir</a></li>
<li class="toctree-l4"><a class="reference internal" href="#trimmer">trimmer</a></li>
<li class="toctree-l4"><a class="reference internal" href="#technical-replicates">technical_replicates</a></li>
<li class="toctree-l4"><a class="reference internal" href="#biological-replicates">biological_replicates</a></li>
<li class="toctree-l4"><a class="reference internal" href="#provider">provider</a></li>
<li class="toctree-l4"><a class="reference internal" href="#aligner">aligner</a></li>
<li class="toctree-l4"><a class="reference internal" href="#samtools-index">samtools_index</a></li>
<li class="toctree-l4"><a class="reference internal" href="#markduplicates">markduplicates</a></li>
<li class="toctree-l4"><a class="reference internal" href="#min-mapping-quality">min_mapping_quality</a></li>
<li class="toctree-l4"><a class="reference internal" href="#tn5-shift">tn5_shift</a></li>
<li class="toctree-l4"><a class="reference internal" href="#remove-blacklist">remove_blacklist</a></li>
<li class="toctree-l4"><a class="reference internal" href="#remove-mito">remove_mito</a></li>
<li class="toctree-l4"><a class="reference internal" href="#only-primary-align">only_primary_align</a></li>
<li class="toctree-l4"><a class="reference internal" href="#create-qc-report">create_qc_report</a></li>
<li class="toctree-l4"><a class="reference internal" href="#bam-sort-mem">bam_sort_mem</a></li>
<li class="toctree-l4"><a class="reference internal" href="#custom-genome-extension">custom_genome_extension</a></li>
<li class="toctree-l4"><a class="reference internal" href="#custom-annotation-extension">custom_annotation_extension</a></li>
<li class="toctree-l4"><a class="reference internal" href="#custom-assembly-suffix">custom_assembly_suffix</a></li>
<li class="toctree-l4"><a class="reference internal" href="#deeptools-qc">deeptools_qc</a></li>
<li class="toctree-l4"><a class="reference internal" href="#deeptools-plotcorrelation">deeptools_plotcorrelation</a></li>
<li class="toctree-l4"><a class="reference internal" href="#deeptools-multibamsummary">deeptools_multibamsummary</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#alignment-workflow-specific">alignment workflow specific</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#bam-sorter">bam_sorter</a></li>
<li class="toctree-l4"><a class="reference internal" href="#cram-no-bam">cram_no_bam</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#peak-calling-chip-atac">peak calling (ChIP &amp; ATAC)</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#peak-caller">peak_caller</a></li>
<li class="toctree-l4"><a class="reference internal" href="#macs2-keep-mates">macs2_keep_mates</a></li>
<li class="toctree-l4"><a class="reference internal" href="#peak-windowsize">peak_windowsize</a></li>
<li class="toctree-l4"><a class="reference internal" href="#slop">slop</a></li>
<li class="toctree-l4"><a class="reference internal" href="#logbase">logbase</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#gene-expression-rna-seq">gene expression (RNA-seq)</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#counts-dir">counts_dir</a></li>
<li class="toctree-l4"><a class="reference internal" href="#dge-dir">dge_dir</a></li>
<li class="toctree-l4"><a class="reference internal" href="#quantifier">quantifier</a></li>
<li class="toctree-l4"><a class="reference internal" href="#tximeta">tximeta</a></li>
<li class="toctree-l4"><a class="reference internal" href="#ignore-strandedness">ignore_strandedness</a></li>
<li class="toctree-l4"><a class="reference internal" href="#dexseq">dexseq</a></li>
<li class="toctree-l4"><a class="reference internal" href="#deseq2">deseq2</a></li>
<li class="toctree-l4"><a class="reference internal" href="#contrasts">contrasts</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#single-cell-atac">single-cell ATAC</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#snaptools-opt">snaptools_opt</a></li>
<li class="toctree-l4"><a class="reference internal" href="#bin-opt">bin_opt</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#single-cell-rna">single-cell RNA</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#fastq-clean-dir">fastq_clean_dir</a></li>
<li class="toctree-l4"><a class="reference internal" href="#fastq-pair">fastq-pair</a></li>
<li class="toctree-l4"><a class="reference internal" href="#id1">quantifier</a></li>
<li class="toctree-l4"><a class="reference internal" href="#barcodefile">barcodefile</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#trackhub">trackhub</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#create-trackhub">create_trackhub</a></li>
<li class="toctree-l4"><a class="reference internal" href="#bigwig-dir">bigwig_dir</a></li>
<li class="toctree-l4"><a class="reference internal" href="#deeptools-flags">deeptools_flags</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="CHANGELOG.html">Changelog</a></li>
<li class="toctree-l2"><a class="reference internal" href="../CONTRIBUTING.html">Contributing to seq2science</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="extracontent.html">Extra resources</a></li>
<li class="toctree-l1"><a class="reference internal" href="faq.html">Frequently Asked Questions (FAQ)</a></li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">seq2science</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          

















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../index.html" class="icon icon-home"></a> &raquo;</li>
        
          <li><a href="extensive_docs.html">Extensive docs</a> &raquo;</li>
        
      <li>All configurable options</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
          
            
              <a href="https://github.com/vanheeringen-lab/seq2science/blob/master/docs/content/schemas.md" class="fa fa-github"> Edit on GitHub</a>
            
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="all-configurable-options">
<h1>All configurable options<a class="headerlink" href="#all-configurable-options" title="Permalink to this headline">¶</a></h1>
<p>This is an automatically generated summary of all configurable options for seq2science. These options are loosely grouped around workflows/topics, however they are generally also shared across workflows. So it is possible that  tunable configuration settings are not mentioned in their topic. At the start of each seq2science run the complete configuration is printed to stdout. You can use that printed configuration as the complete list of tunable configuration settings.</p>
<p>We believe that all our default settings are reasonable, and manual finetuning is generally not required.</p>
<div class="section" id="general">
<h2>general<a class="headerlink" href="#general" title="Permalink to this headline">¶</a></h2>
<div class="section" id="samples">
<h3>samples<a class="headerlink" href="#samples" title="Permalink to this headline">¶</a></h3>
<div class="highlight-guess notranslate"><div class="highlight"><pre><span></span><span class="c">samples:</span>
<span class="c">    description: tab-separated file specifying the samples to run through the workflow</span>
<span class="c">    default: samples.tsv</span>
<span class="c">    type: string</span>
</pre></div>
</div>
</div>
<div class="section" id="rule-dir">
<h3>rule_dir<a class="headerlink" href="#rule-dir" title="Permalink to this headline">¶</a></h3>
<div class="highlight-guess notranslate"><div class="highlight"><pre><span></span><span class="c">rule_dir:</span>
<span class="c">    description: directory where to find the workflow rules</span>
<span class="c">    default: ../../rules</span>
<span class="c">    type: string</span>
</pre></div>
</div>
</div>
<div class="section" id="result-dir">
<h3>result_dir<a class="headerlink" href="#result-dir" title="Permalink to this headline">¶</a></h3>
<div class="highlight-guess notranslate"><div class="highlight"><pre><span></span><span class="c">result_dir:</span>
<span class="c">    description: where to store the pipeline&#39;s results (by default)</span>
<span class="c">    default: ./</span>
<span class="c">    type: string</span>
</pre></div>
</div>
</div>
<div class="section" id="genome-dir">
<h3>genome_dir<a class="headerlink" href="#genome-dir" title="Permalink to this headline">¶</a></h3>
<div class="highlight-guess notranslate"><div class="highlight"><pre><span></span><span class="c">genome_dir:</span>
<span class="c">    description: where to store the pipeline&#39;s genome assemblies</span>
<span class="c">    default: ./</span>
<span class="c">    type: string</span>
</pre></div>
</div>
</div>
<div class="section" id="log-dir">
<h3>log_dir<a class="headerlink" href="#log-dir" title="Permalink to this headline">¶</a></h3>
<div class="highlight-guess notranslate"><div class="highlight"><pre><span></span><span class="c">log_dir:</span>
<span class="c">    description: directory where to store the logs (defaults inside the result dir)</span>
<span class="c">    default: log</span>
<span class="c">    type: string</span>
</pre></div>
</div>
</div>
<div class="section" id="benchmark-dir">
<h3>benchmark_dir<a class="headerlink" href="#benchmark-dir" title="Permalink to this headline">¶</a></h3>
<div class="highlight-guess notranslate"><div class="highlight"><pre><span></span><span class="c">benchmark_dir:</span>
<span class="c">    description: directory where to store the logs (defaults inside the result dir)</span>
<span class="c">    default: benchmark</span>
<span class="c">    type: string</span>
</pre></div>
</div>
</div>
<div class="section" id="fqext1">
<h3>fqext1<a class="headerlink" href="#fqext1" title="Permalink to this headline">¶</a></h3>
<div class="highlight-guess notranslate"><div class="highlight"><pre><span></span><span class="c">fqext1:</span>
<span class="c">    description: filename suffix when handling paired-end data, describing the forward read</span>
<span class="c">    default: R1</span>
<span class="c">    type: string</span>
</pre></div>
</div>
</div>
<div class="section" id="fqext2">
<h3>fqext2<a class="headerlink" href="#fqext2" title="Permalink to this headline">¶</a></h3>
<div class="highlight-guess notranslate"><div class="highlight"><pre><span></span><span class="c">fqext2:</span>
<span class="c">    description: filename suffix when handling paired-end data, describing the reverse read</span>
<span class="c">    default: R2</span>
<span class="c">    type: string</span>
</pre></div>
</div>
</div>
<div class="section" id="fqsuffix">
<h3>fqsuffix<a class="headerlink" href="#fqsuffix" title="Permalink to this headline">¶</a></h3>
<div class="highlight-guess notranslate"><div class="highlight"><pre><span></span><span class="c">fqsuffix:</span>
<span class="c">    description: file descriptor for fastq files (often fastq or fq)</span>
<span class="c">    default: fastq</span>
<span class="c">    type: string</span>
</pre></div>
</div>
</div>
<div class="section" id="cpulimit">
<h3>cpulimit<a class="headerlink" href="#cpulimit" title="Permalink to this headline">¶</a></h3>
<div class="highlight-guess notranslate"><div class="highlight"><pre><span></span><span class="c">cpulimit:</span>
<span class="c">    description: whether or not to make use of cpulimit to enfoce max thread usage for some rules</span>
<span class="c">    default: True</span>
<span class="c">    type: boolean</span>
</pre></div>
</div>
</div>
<div class="section" id="email">
<h3>email<a class="headerlink" href="#email" title="Permalink to this headline">¶</a></h3>
<div class="highlight-guess notranslate"><div class="highlight"><pre><span></span><span class="c">email:</span>
<span class="c">    description: email to reach you after pipeline finished, required for trackhub and multiqc report</span>
<span class="c">    type: string</span>
</pre></div>
</div>
</div>
</div>
<div class="section" id="alignment-general">
<h2>alignment general<a class="headerlink" href="#alignment-general" title="Permalink to this headline">¶</a></h2>
<div class="section" id="trimmed-dir">
<h3>trimmed_dir<a class="headerlink" href="#trimmed-dir" title="Permalink to this headline">¶</a></h3>
<div class="highlight-guess notranslate"><div class="highlight"><pre><span></span><span class="c">trimmed_dir:</span>
<span class="c">    description: directory where to store the workflow trimmed fastqs (defaults inside the result dir)</span>
<span class="c">    default: fastq_trimmed</span>
<span class="c">    type: string</span>
</pre></div>
</div>
</div>
<div class="section" id="qc-dir">
<h3>qc_dir<a class="headerlink" href="#qc-dir" title="Permalink to this headline">¶</a></h3>
<div class="highlight-guess notranslate"><div class="highlight"><pre><span></span><span class="c">qc_dir:</span>
<span class="c">    description: directory where to store the workflow quality controls (defaults inside the result dir)</span>
<span class="c">    default: qc</span>
<span class="c">    type: string</span>
</pre></div>
</div>
</div>
<div class="section" id="final-bam-dir">
<h3>final_bam_dir<a class="headerlink" href="#final-bam-dir" title="Permalink to this headline">¶</a></h3>
<div class="highlight-guess notranslate"><div class="highlight"><pre><span></span><span class="c">final_bam_dir:</span>
<span class="c">    description: directory where to store the workflow&#39;s final deduplicated &amp; sieved bams (defaults inside the result dir)</span>
<span class="c">    default: final_bam</span>
<span class="c">    type: string</span>
</pre></div>
</div>
</div>
<div class="section" id="trimmer">
<h3>trimmer<a class="headerlink" href="#trimmer" title="Permalink to this headline">¶</a></h3>
<div class="highlight-guess notranslate"><div class="highlight"><pre><span></span><span class="c">trimmer:</span>
<span class="c">    description: which adapter trimmer to use</span>
<span class="c">    properties:</span>
<span class="c">        trimgalore:</span>
<span class="c">            description: trim galore! settings (note that the adapter is autodetected, so you do not have to set it here)</span>
<span class="c">            properties:</span>
<span class="c">                trimoptions:</span>
<span class="c">                    default: --quality 10 --length 20 --consider_already_trimmed 10</span>
<span class="c">            default:</span>
<span class="c">                trimoptions: --quality 10 --length 20 --consider_already_trimmed 10</span>
<span class="c">        fastp:</span>
<span class="c">            description: fastp settings (note that the adapter is autodetected, so you do not have to set it here)</span>
<span class="c">            properties:</span>
<span class="c">                trimoptions:</span>
<span class="c">                    default:</span>
<span class="c">            default:</span>
<span class="c">                trimoptions:</span>
<span class="c">    default:</span>
<span class="c">        fastp:</span>
<span class="c">            trimoptions:</span>
<span class="c">    type: object</span>
</pre></div>
</div>
</div>
<div class="section" id="technical-replicates">
<h3>technical_replicates<a class="headerlink" href="#technical-replicates" title="Permalink to this headline">¶</a></h3>
<div class="highlight-guess notranslate"><div class="highlight"><pre><span></span><span class="c">technical_replicates:</span>
<span class="c">    description: concatenates samples depending on the names given in the &#39;replicates&#39; column of samples.tsv</span>
<span class="c">    default: merge</span>
<span class="c">    enum: [&#39;merge&#39;, &#39;keep&#39;]</span>
</pre></div>
</div>
</div>
<div class="section" id="biological-replicates">
<h3>biological_replicates<a class="headerlink" href="#biological-replicates" title="Permalink to this headline">¶</a></h3>
<div class="highlight-guess notranslate"><div class="highlight"><pre><span></span><span class="c">biological_replicates:</span>
<span class="c">    description: which method to use to combine replicates (fisher (n &gt;=2), idr (n==2), or keep)</span>
<span class="c">    default: keep</span>
<span class="c">    enum: [&#39;fisher&#39;, &#39;idr&#39;, &#39;keep&#39;]</span>
<span class="c">    type: string</span>
</pre></div>
</div>
</div>
<div class="section" id="provider">
<h3>provider<a class="headerlink" href="#provider" title="Permalink to this headline">¶</a></h3>
<div class="highlight-guess notranslate"><div class="highlight"><pre><span></span><span class="c">provider:</span>
<span class="c">    description: Specify provider to download new genomes from. Will try Ensembl &gt; UCSC &gt; NCBI if left blank.</span>
<span class="c">    enum: [&#39;Ensembl&#39;, &#39;UCSC&#39;, &#39;NCBI&#39;]</span>
<span class="c">    type: string</span>
</pre></div>
</div>
</div>
<div class="section" id="aligner">
<h3>aligner<a class="headerlink" href="#aligner" title="Permalink to this headline">¶</a></h3>
<div class="highlight-guess notranslate"><div class="highlight"><pre><span></span><span class="c">aligner:</span>
<span class="c">    description: which aligner to use</span>
<span class="c">    properties:</span>
<span class="c">        bowtie2:</span>
<span class="c">            properties:</span>
<span class="c">                index:</span>
<span class="c">                    default:</span>
<span class="c">                align:</span>
<span class="c">                    default:</span>
<span class="c">            default:</span>
<span class="c">                index:</span>
<span class="c">                align:</span>
<span class="c">        bwa-mem:</span>
<span class="c">            properties:</span>
<span class="c">                index:</span>
<span class="c">                    default: -a bwtsw</span>
<span class="c">                align:</span>
<span class="c">                    default: -M</span>
<span class="c">            default:</span>
<span class="c">                index: -a bwtsw</span>
<span class="c">                align: -M</span>
<span class="c">        bwa-mem2:</span>
<span class="c">            properties:</span>
<span class="c">                index:</span>
<span class="c">                    default:</span>
<span class="c">                align:</span>
<span class="c">                    default: -M</span>
<span class="c">            default:</span>
<span class="c">                index:</span>
<span class="c">                align:</span>
<span class="c">        hisat2:</span>
<span class="c">            properties:</span>
<span class="c">                index:</span>
<span class="c">                    default:</span>
<span class="c">                align:</span>
<span class="c">                    default:</span>
<span class="c">            default:</span>
<span class="c">                index:</span>
<span class="c">                align:</span>
<span class="c">        minimap2:</span>
<span class="c">            properties:</span>
<span class="c">                index:</span>
<span class="c">                    default:</span>
<span class="c">                align:</span>
<span class="c">                    default: -a sr --secondary=no</span>
<span class="c">            default:</span>
<span class="c">                index:</span>
<span class="c">                align:</span>
<span class="c">        star:</span>
<span class="c">            properties:</span>
<span class="c">                index:</span>
<span class="c">                    default: --limitGenomeGenerateRAM 37000000000 --genomeSAsparseD 1</span>
<span class="c">                align:</span>
<span class="c">                    default:</span>
<span class="c">            default:</span>
<span class="c">                index: --limitGenomeGenerateRAM 37000000000 --genomeSAsparseD 1</span>
<span class="c">                align:</span>
<span class="c">    default:</span>
<span class="c">        bwa-mem2:</span>
<span class="c">            index:</span>
<span class="c">            align:</span>
</pre></div>
</div>
</div>
<div class="section" id="samtools-index">
<h3>samtools_index<a class="headerlink" href="#samtools-index" title="Permalink to this headline">¶</a></h3>
<div class="highlight-guess notranslate"><div class="highlight"><pre><span></span><span class="c">samtools_index:</span>
<span class="c">    description: samtools index settings</span>
<span class="c">    default:</span>
<span class="c">    type: string</span>
</pre></div>
</div>
</div>
<div class="section" id="markduplicates">
<h3>markduplicates<a class="headerlink" href="#markduplicates" title="Permalink to this headline">¶</a></h3>
<div class="highlight-guess notranslate"><div class="highlight"><pre><span></span><span class="c">markduplicates:</span>
<span class="c">    description: the parameters of picard markduplicates</span>
<span class="c">    default: REMOVE_DUPLICATES=true -Xms4G -Xmx6G MAX_FILE_HANDLES_FOR_READ_ENDS_MAP=999</span>
<span class="c">    type: string</span>
</pre></div>
</div>
</div>
<div class="section" id="min-mapping-quality">
<h3>min_mapping_quality<a class="headerlink" href="#min-mapping-quality" title="Permalink to this headline">¶</a></h3>
<div class="highlight-guess notranslate"><div class="highlight"><pre><span></span><span class="c">min_mapping_quality:</span>
<span class="c">    description: minimum mapq to keep aligned reads</span>
<span class="c">    default: 0</span>
<span class="c">    type: integer</span>
</pre></div>
</div>
</div>
<div class="section" id="tn5-shift">
<h3>tn5_shift<a class="headerlink" href="#tn5-shift" title="Permalink to this headline">¶</a></h3>
<div class="highlight-guess notranslate"><div class="highlight"><pre><span></span><span class="c">tn5_shift:</span>
<span class="c">    description: whether or not to shift reads to adjust for the tn5 bias</span>
<span class="c">    default: False</span>
<span class="c">    type: boolean</span>
</pre></div>
</div>
</div>
<div class="section" id="remove-blacklist">
<h3>remove_blacklist<a class="headerlink" href="#remove-blacklist" title="Permalink to this headline">¶</a></h3>
<div class="highlight-guess notranslate"><div class="highlight"><pre><span></span><span class="c">remove_blacklist:</span>
<span class="c">    description: whether or not to remove reads in the encode blacklisted regions</span>
<span class="c">    default: False</span>
<span class="c">    type: boolean</span>
</pre></div>
</div>
</div>
<div class="section" id="remove-mito">
<h3>remove_mito<a class="headerlink" href="#remove-mito" title="Permalink to this headline">¶</a></h3>
<div class="highlight-guess notranslate"><div class="highlight"><pre><span></span><span class="c">remove_mito:</span>
<span class="c">    description: whether or not to remove reads aligned to the mitochondria</span>
<span class="c">    default: False</span>
<span class="c">    type: boolean</span>
</pre></div>
</div>
</div>
<div class="section" id="only-primary-align">
<h3>only_primary_align<a class="headerlink" href="#only-primary-align" title="Permalink to this headline">¶</a></h3>
<div class="highlight-guess notranslate"><div class="highlight"><pre><span></span><span class="c">only_primary_align:</span>
<span class="c">    description: whether or not to only keep primary aligned reads (no multimappers)</span>
<span class="c">    default: False</span>
<span class="c">    type: boolean</span>
</pre></div>
</div>
</div>
<div class="section" id="create-qc-report">
<h3>create_qc_report<a class="headerlink" href="#create-qc-report" title="Permalink to this headline">¶</a></h3>
<div class="highlight-guess notranslate"><div class="highlight"><pre><span></span><span class="c">create_qc_report:</span>
<span class="c">    description: whether or not to generate a multiqc report</span>
<span class="c">    default: True</span>
<span class="c">    type: boolean</span>
</pre></div>
</div>
</div>
<div class="section" id="bam-sort-mem">
<h3>bam_sort_mem<a class="headerlink" href="#bam-sort-mem" title="Permalink to this headline">¶</a></h3>
<div class="highlight-guess notranslate"><div class="highlight"><pre><span></span><span class="c">bam_sort_mem:</span>
<span class="c">    description: The max memory used in gigabytes by samtools/sambamba when sorting bams</span>
<span class="c">    default: 2</span>
<span class="c">    type: integer</span>
</pre></div>
</div>
</div>
<div class="section" id="custom-genome-extension">
<h3>custom_genome_extension<a class="headerlink" href="#custom-genome-extension" title="Permalink to this headline">¶</a></h3>
<div class="highlight-guess notranslate"><div class="highlight"><pre><span></span><span class="c">custom_genome_extension:</span>
<span class="c">    description: File(s) to append to the genome (must be in fasta format)</span>
</pre></div>
</div>
</div>
<div class="section" id="custom-annotation-extension">
<h3>custom_annotation_extension<a class="headerlink" href="#custom-annotation-extension" title="Permalink to this headline">¶</a></h3>
<div class="highlight-guess notranslate"><div class="highlight"><pre><span></span><span class="c">custom_annotation_extension:</span>
<span class="c">    description: File(s) to append to the gene annotations (must be in GTF format)</span>
</pre></div>
</div>
</div>
<div class="section" id="custom-assembly-suffix">
<h3>custom_assembly_suffix<a class="headerlink" href="#custom-assembly-suffix" title="Permalink to this headline">¶</a></h3>
<div class="highlight-guess notranslate"><div class="highlight"><pre><span></span><span class="c">custom_assembly_suffix:</span>
<span class="c">    description: What to append to the assembly name to indicate that custom data has been added to either genome or assembly.</span>
<span class="c">    default: _custom</span>
<span class="c">    type: string</span>
</pre></div>
</div>
</div>
<div class="section" id="deeptools-qc">
<h3>deeptools_qc<a class="headerlink" href="#deeptools-qc" title="Permalink to this headline">¶</a></h3>
<div class="highlight-guess notranslate"><div class="highlight"><pre><span></span><span class="c">deeptools_qc:</span>
<span class="c">    description: Whether or not to perform deeptools QC. With many samples this can take a long time and might be undesirable</span>
<span class="c">    default: True</span>
<span class="c">    type: boolean</span>
</pre></div>
</div>
</div>
<div class="section" id="deeptools-plotcorrelation">
<h3>deeptools_plotcorrelation<a class="headerlink" href="#deeptools-plotcorrelation" title="Permalink to this headline">¶</a></h3>
<div class="highlight-guess notranslate"><div class="highlight"><pre><span></span><span class="c">deeptools_plotcorrelation:</span>
<span class="c">    description: Deeptools plotCorrelation params</span>
<span class="c">    default: --colorMap RdYlBu_r --plotNumbers</span>
<span class="c">    type: string</span>
</pre></div>
</div>
</div>
<div class="section" id="deeptools-multibamsummary">
<h3>deeptools_multibamsummary<a class="headerlink" href="#deeptools-multibamsummary" title="Permalink to this headline">¶</a></h3>
<div class="highlight-guess notranslate"><div class="highlight"><pre><span></span><span class="c">deeptools_multibamsummary:</span>
<span class="c">    description: Deeptools multiBamSummary params</span>
<span class="c">    default: --distanceBetweenBins 9000 --binSize 1000</span>
<span class="c">    type: string</span>
</pre></div>
</div>
</div>
</div>
<div class="section" id="alignment-workflow-specific">
<h2>alignment workflow specific<a class="headerlink" href="#alignment-workflow-specific" title="Permalink to this headline">¶</a></h2>
<div class="section" id="bam-sorter">
<h3>bam_sorter<a class="headerlink" href="#bam-sorter" title="Permalink to this headline">¶</a></h3>
<div class="highlight-guess notranslate"><div class="highlight"><pre><span></span><span class="c">bam_sorter:</span>
<span class="c">    description: which sorting software to use</span>
<span class="c">    type: object</span>
<span class="c">    properties:</span>
<span class="c">        description: sort order of aligned reads</span>
<span class="c">        default: coordinate</span>
<span class="c">        enum: [&#39;queryname&#39;, &#39;coordinate&#39;]</span>
<span class="c">        type: string</span>
<span class="c">    default:</span>
<span class="c">        samtools: coordinate</span>
</pre></div>
</div>
</div>
<div class="section" id="cram-no-bam">
<h3>cram_no_bam<a class="headerlink" href="#cram-no-bam" title="Permalink to this headline">¶</a></h3>
<div class="highlight-guess notranslate"><div class="highlight"><pre><span></span><span class="c">cram_no_bam:</span>
<span class="c">    description: whether to store cram instead of bam</span>
<span class="c">    type: boolean</span>
<span class="c">    default: False</span>
</pre></div>
</div>
</div>
</div>
<div class="section" id="peak-calling-chip-atac">
<h2>peak calling (ChIP &amp; ATAC)<a class="headerlink" href="#peak-calling-chip-atac" title="Permalink to this headline">¶</a></h2>
<div class="section" id="peak-caller">
<h3>peak_caller<a class="headerlink" href="#peak-caller" title="Permalink to this headline">¶</a></h3>
<div class="highlight-guess notranslate"><div class="highlight"><pre><span></span><span class="c">peak_caller:</span>
<span class="c">    type: object</span>
<span class="c">    properties:</span>
<span class="c">        genrich:</span>
<span class="c">            type: string</span>
<span class="c">        hmmratac:</span>
<span class="c">            type: string</span>
<span class="c">        macs2:</span>
<span class="c">            type: string</span>
<span class="c">    minProperties: 1</span>
<span class="c">    additionalProperties: False</span>
<span class="c">    default:</span>
<span class="c">        macs2: --shift -100 --extsize 200 --nomodel --keep-dup 1 --buffer-size 10000</span>
</pre></div>
</div>
</div>
<div class="section" id="macs2-keep-mates">
<h3>macs2_keep_mates<a class="headerlink" href="#macs2-keep-mates" title="Permalink to this headline">¶</a></h3>
<div class="highlight-guess notranslate"><div class="highlight"><pre><span></span><span class="c">macs2_keep_mates:</span>
<span class="c">    description: macs2 normally removes paired-end mates when doing --nomodel shift peak calling. When this flag is on, the paired-end bam is converted to sngle-end bam, so all reads are counted. This is not conventional to do, but might help with low sequencing depth.</span>
<span class="c">    default: False</span>
<span class="c">    type: boolean</span>
</pre></div>
</div>
</div>
<div class="section" id="peak-windowsize">
<h3>peak_windowsize<a class="headerlink" href="#peak-windowsize" title="Permalink to this headline">¶</a></h3>
<div class="highlight-guess notranslate"><div class="highlight"><pre><span></span><span class="c">peak_windowsize:</span>
<span class="c">    description: when combining narrowpeaks, how much each summit should be extended by when considering overlapping peaks. A summit is expanded by the windowsize on each side, so the total width is twice the peak_windowsize.</span>
<span class="c">    default: 100</span>
<span class="c">    type: integer</span>
</pre></div>
</div>
</div>
<div class="section" id="slop">
<h3>slop<a class="headerlink" href="#slop" title="Permalink to this headline">¶</a></h3>
<div class="highlight-guess notranslate"><div class="highlight"><pre><span></span><span class="c">slop:</span>
<span class="c">    description: how much the summits should be extended by (on each side) for making a peak count table. When for instance a slop of 100 is chosen, the final peak&#39;s width is 200.</span>
<span class="c">    default: 100</span>
<span class="c">    type: integer</span>
</pre></div>
</div>
</div>
<div class="section" id="logbase">
<h3>logbase<a class="headerlink" href="#logbase" title="Permalink to this headline">¶</a></h3>
<div class="highlight-guess notranslate"><div class="highlight"><pre><span></span><span class="c">logbase:</span>
<span class="c">    description: The log base of the number of reads under peak count tables</span>
<span class="c">    default: 2</span>
<span class="c">    type: number</span>
</pre></div>
</div>
</div>
</div>
<div class="section" id="gene-expression-rna-seq">
<h2>gene expression (RNA-seq)<a class="headerlink" href="#gene-expression-rna-seq" title="Permalink to this headline">¶</a></h2>
<div class="section" id="counts-dir">
<h3>counts_dir<a class="headerlink" href="#counts-dir" title="Permalink to this headline">¶</a></h3>
<div class="highlight-guess notranslate"><div class="highlight"><pre><span></span><span class="c">counts_dir:</span>
<span class="c">    description: directory where to store the gene counts for DGE analysis (defaults inside the result dir)</span>
<span class="c">    default: gene_counts</span>
<span class="c">    type: string</span>
</pre></div>
</div>
</div>
<div class="section" id="dge-dir">
<h3>dge_dir<a class="headerlink" href="#dge-dir" title="Permalink to this headline">¶</a></h3>
<div class="highlight-guess notranslate"><div class="highlight"><pre><span></span><span class="c">dge_dir:</span>
<span class="c">    description: directory where to store the workflow differential gene analysis output (defaults inside the result dir)</span>
<span class="c">    default: deseq2</span>
<span class="c">    type: string</span>
</pre></div>
</div>
</div>
<div class="section" id="quantifier">
<h3>quantifier<a class="headerlink" href="#quantifier" title="Permalink to this headline">¶</a></h3>
<div class="highlight-guess notranslate"><div class="highlight"><pre><span></span><span class="c">quantifier:</span>
<span class="c">    description: which quantifier to use</span>
<span class="c">    properties:</span>
<span class="c">        salmon:</span>
<span class="c">            properties:</span>
<span class="c">                quantifier_index:</span>
<span class="c">                    default: -k 31</span>
<span class="c">                quantifier_flags:</span>
<span class="c">                    default: --seqBias --gcBias --validateMappings --recoverOrphans</span>
<span class="c">                decoy_aware_index:</span>
<span class="c">                    default: False</span>
<span class="c">            default:</span>
<span class="c">                quantifier_index: -k 31</span>
<span class="c">                quantifier_flags: --seqBias --gcBias --validateMappings --recoverOrphans</span>
<span class="c">                decoy_aware_index: False</span>
<span class="c">        htseq:</span>
<span class="c">            properties:</span>
<span class="c">                htseq_flags:</span>
<span class="c">                    default:</span>
<span class="c">            default:</span>
<span class="c">                htseq_flags:</span>
<span class="c">        featurecounts:</span>
<span class="c">            properties:</span>
<span class="c">                featurecounts_flags:</span>
<span class="c">                    default:</span>
<span class="c">            default:</span>
<span class="c">                featurecounts_flags:</span>
<span class="c">    default:</span>
<span class="c">        htseq:</span>
<span class="c">            htseq_flags:</span>
</pre></div>
</div>
</div>
<div class="section" id="tximeta">
<h3>tximeta<a class="headerlink" href="#tximeta" title="Permalink to this headline">¶</a></h3>
<div class="highlight-guess notranslate"><div class="highlight"><pre><span></span><span class="c">tximeta:</span>
<span class="c">    description: tximeta is used to convert salmon output (estimated transcript abundances) to a gene counts matrix. Currently, these values can be ignored.</span>
<span class="c">    properties:</span>
<span class="c">        source:</span>
<span class="c">            default: Ensembl</span>
<span class="c">        organism:</span>
<span class="c">            default: tximeta_file</span>
<span class="c">        release:</span>
<span class="c">            default: 42</span>
<span class="c">    default:</span>
<span class="c">        source: Ensembl</span>
<span class="c">        organism: tximeta_file</span>
<span class="c">        release: 42</span>
</pre></div>
</div>
</div>
<div class="section" id="ignore-strandedness">
<h3>ignore_strandedness<a class="headerlink" href="#ignore-strandedness" title="Permalink to this headline">¶</a></h3>
<div class="highlight-guess notranslate"><div class="highlight"><pre><span></span><span class="c">ignore_strandedness:</span>
<span class="c">    description: ignore/dont infer strand-specificity?</span>
<span class="c">    default: False</span>
<span class="c">    type: boolean</span>
</pre></div>
</div>
</div>
<div class="section" id="dexseq">
<h3>dexseq<a class="headerlink" href="#dexseq" title="Permalink to this headline">¶</a></h3>
<div class="highlight-guess notranslate"><div class="highlight"><pre><span></span><span class="c">dexseq:</span>
<span class="c">    description: output an exon counts matrix for use in DEXSeq.</span>
<span class="c">    default: False</span>
<span class="c">    type: boolean</span>
</pre></div>
</div>
</div>
<div class="section" id="deseq2">
<h3>deseq2<a class="headerlink" href="#deseq2" title="Permalink to this headline">¶</a></h3>
<div class="highlight-guess notranslate"><div class="highlight"><pre><span></span><span class="c">deseq2:</span>
<span class="c">    description: DESeq2 settings. See the DESeq2 vignette for details</span>
<span class="c">    properties:</span>
<span class="c">        multiple_testing_procedure:</span>
<span class="c">            default: BH</span>
<span class="c">            enum: [&#39;BH&#39;, &#39;IHW&#39;]</span>
<span class="c">            type: string</span>
<span class="c">        alpha_value:</span>
<span class="c">            default: 0.1</span>
<span class="c">            type: number</span>
<span class="c">        shrinkage_estimator:</span>
<span class="c">            default: apeglm</span>
<span class="c">            enum: [&#39;apeglm&#39;, &#39;ashr&#39;, &#39;normal&#39;]</span>
<span class="c">            type: string</span>
<span class="c">    default:</span>
<span class="c">        multiple_testing_procedure: BH</span>
<span class="c">        alpha_value: 0.1</span>
<span class="c">        shrinkage_estimator: apeglm</span>
</pre></div>
</div>
</div>
<div class="section" id="contrasts">
<h3>contrasts<a class="headerlink" href="#contrasts" title="Permalink to this headline">¶</a></h3>
<div class="highlight-guess notranslate"><div class="highlight"><pre><span></span><span class="c">contrasts:</span>
<span class="c">    description: deseq2 design contrast(s)</span>
</pre></div>
</div>
</div>
</div>
<div class="section" id="single-cell-atac">
<h2>single-cell ATAC<a class="headerlink" href="#single-cell-atac" title="Permalink to this headline">¶</a></h2>
<div class="section" id="snaptools-opt">
<h3>snaptools_opt<a class="headerlink" href="#snaptools-opt" title="Permalink to this headline">¶</a></h3>
<div class="highlight-guess notranslate"><div class="highlight"><pre><span></span><span class="c">snaptools_opt:</span>
<span class="c">    description: snaptools optional arguments for snaptools snap-pre, from snaptools (https://github.com/r3fang/SnapTools). For more info run snaptools snap-pre -h</span>
<span class="c">    default: --min-flen=0 --max-flen=1000 --keep-single=FALSE --keep-secondary=FALSE --overwrite=True --min-cov=100 --verbose=True</span>
</pre></div>
</div>
</div>
<div class="section" id="bin-opt">
<h3>bin_opt<a class="headerlink" href="#bin-opt" title="Permalink to this headline">¶</a></h3>
<div class="highlight-guess notranslate"><div class="highlight"><pre><span></span><span class="c">bin_opt:</span>
<span class="c">    description: snaptools optional arguments for snaptools snap-add-bmat, from snaptools (https://github.com/r3fang/SnapTools). The genome will be divided into bins of the equal size of --bin-size-list to create the cell x bin count matrix. For more info run snaptools snap-add-bmat -h</span>
<span class="c">    default: --bin-size-list 5000 --verbose=True</span>
</pre></div>
</div>
</div>
</div>
<div class="section" id="single-cell-rna">
<h2>single-cell RNA<a class="headerlink" href="#single-cell-rna" title="Permalink to this headline">¶</a></h2>
<div class="section" id="fastq-clean-dir">
<h3>fastq_clean_dir<a class="headerlink" href="#fastq-clean-dir" title="Permalink to this headline">¶</a></h3>
<div class="highlight-guess notranslate"><div class="highlight"><pre><span></span><span class="c">fastq_clean_dir:</span>
<span class="c">    description: directory where to find the workflow rules</span>
<span class="c">    default: fastq_cleaned</span>
<span class="c">    type: string</span>
</pre></div>
</div>
</div>
<div class="section" id="fastq-pair">
<h3>fastq-pair<a class="headerlink" href="#fastq-pair" title="Permalink to this headline">¶</a></h3>
<div class="highlight-guess notranslate"><div class="highlight"><pre><span></span><span class="c">fastq-pair:</span>
<span class="c">    description: fastq synchronization</span>
<span class="c">    default:</span>
</pre></div>
</div>
</div>
<div class="section" id="id1">
<h3>quantifier<a class="headerlink" href="#id1" title="Permalink to this headline">¶</a></h3>
<div class="highlight-guess notranslate"><div class="highlight"><pre><span></span><span class="c">quantifier:</span>
<span class="c">    description: which quantifier to use</span>
<span class="c">    properties:</span>
<span class="c">        kallistobus:</span>
<span class="c">            properties:</span>
<span class="c">                ref:</span>
<span class="c">                    default:</span>
<span class="c">                count:</span>
<span class="c">                    default: -x 10XV3 --verbose --h5ad --workflow lamanno</span>
<span class="c">    default:</span>
<span class="c">        kallistobus:</span>
<span class="c">            ref:</span>
<span class="c">            count: -x 10XV3 --verbose --h5ad --workflow lamanno</span>
</pre></div>
</div>
</div>
<div class="section" id="barcodefile">
<h3>barcodefile<a class="headerlink" href="#barcodefile" title="Permalink to this headline">¶</a></h3>
<div class="highlight-guess notranslate"><div class="highlight"><pre><span></span><span class="c">barcodefile:</span>
<span class="c">    type: string</span>
</pre></div>
</div>
</div>
</div>
<div class="section" id="trackhub">
<h2>trackhub<a class="headerlink" href="#trackhub" title="Permalink to this headline">¶</a></h2>
<div class="section" id="create-trackhub">
<h3>create_trackhub<a class="headerlink" href="#create-trackhub" title="Permalink to this headline">¶</a></h3>
<div class="highlight-guess notranslate"><div class="highlight"><pre><span></span><span class="c">create_trackhub:</span>
<span class="c">    description: generate a trackhub (and required files)?</span>
<span class="c">    default: True</span>
<span class="c">    type: boolean</span>
</pre></div>
</div>
</div>
<div class="section" id="bigwig-dir">
<h3>bigwig_dir<a class="headerlink" href="#bigwig-dir" title="Permalink to this headline">¶</a></h3>
<div class="highlight-guess notranslate"><div class="highlight"><pre><span></span><span class="c">bigwig_dir:</span>
<span class="c">    description: directory where to store the bigwigs used by the trackhub (defaults inside the result dir)</span>
<span class="c">    default: bigwigs</span>
<span class="c">    type: string</span>
</pre></div>
</div>
</div>
<div class="section" id="deeptools-flags">
<h3>deeptools_flags<a class="headerlink" href="#deeptools-flags" title="Permalink to this headline">¶</a></h3>
<div class="highlight-guess notranslate"><div class="highlight"><pre><span></span><span class="c">deeptools_flags:</span>
<span class="c">    description: flags for bam to bigwig conversions with deeptools bamCoverage</span>
<span class="c">    default: --normalizeUsing BPM</span>
</pre></div>
</div>
</div>
</div>
</div>


           </div>
           
          </div>
          <footer>
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
        <a href="CHANGELOG.html" class="btn btn-neutral float-right" title="Changelog" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
        <a href="all_rules.html" class="btn btn-neutral float-left" title="Per rule explanation" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>
        &#169; Copyright Maarten van der Sande, Siebren Frölich, Jos Smits, Rebecca Snabel, Tilman Schäfers, &amp; Simon van Heeringen..

    </p>
  </div>
    
    
    
    Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>
        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>